src/libs/ajax/Metrics.ts

```diff
@@ -8,8 +8,8 @@ import { authStore, userStatus } from 'src/libs/state'
 import { v4 as uuid } from 'uuid'
 
 
-export const Metrics = signal => ({
-  captureEvent: withErrorIgnoring(async (event, details = {}) => {
+export const Metrics = (signal?: AbortSignal) => {
+  const captureEventFn = async (event, details = {}) => {
```

Comment:
breaking out this async function improves the Typescript flow out to the consumers